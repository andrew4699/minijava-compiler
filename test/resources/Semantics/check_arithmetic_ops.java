class a {

  public static void main(String[] args) {
    System.out.println(1);
  }
}

class b {
  public int valid() {
    int a;
    int b;
    double d;
    double e;
    // check plus
    a = 1 + 1;
    b = a + 1;
    a = b + 1;
    d = 2.;
    e = .24 + 2e4 + d + 12.;

    // check minus
    a = 1 - 0;
    b = b - 1;
    a = a - b;
    d = d - e - 21.;

    // check times
    a = a * b;
    b = a * 23;
    a = 23 * b;
    d = .1 * 2e4 * e;

    return 1;
  }

  public int invalid() {
    int a;
    boolean b;
    double d;
    double e;
    b = false;
    d = .0;
    e = .0;

    // check plus
    a = 1 + b;
    a = b + 1;
    a = b + b;
    d = 2e4 + a;
    e = 2.4 + b;
    d = a + e + b;

    // check minus
    a = 1 - b;
    b = b - 1;
    b = b - b;
    d = 2e5 - a;
    e = 2.5 - b;
    d = a - e - b;

    // check times
    a = a * a;
    b = b * 1;
    a = 23 * b;
    d = 2e6 * a;
    e = 2.6 * b;
    d = a * e * b;

    return 0;
  }
}
