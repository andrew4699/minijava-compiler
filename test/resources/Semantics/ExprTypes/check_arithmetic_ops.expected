[ExprTypesVisitor - 46]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 46]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 47]: Expression yielded type boolean where int was expected.
[ExprTypesVisitor - 47]: Expression yielded type boolean where int was expected.
[ExprTypesVisitor - 48]: Expression yielded type boolean where int was expected.
[ExprTypesVisitor - 48]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 49]: Expression yielded type double where int was expected.
[ExprTypesVisitor - 50]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 50]: Expression yielded type double where boolean was expected.
[ExprTypesVisitor - 51]: Expression yielded type int where double was expected.
[ExprTypesVisitor - 51]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 51]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 51]: Expression yielded type int where double was expected.
[ExprTypesVisitor - 54]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 54]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 55]: Expression yielded type boolean where int was expected.
[ExprTypesVisitor - 56]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 57]: Expression yielded type double where int was expected.
[ExprTypesVisitor - 58]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 58]: Expression yielded type double where boolean was expected.
[ExprTypesVisitor - 59]: Expression yielded type int where double was expected.
[ExprTypesVisitor - 59]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 59]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 59]: Expression yielded type int where double was expected.
[ExprTypesVisitor - 63]: Expression yielded type boolean where int was expected.
[ExprTypesVisitor - 64]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 64]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 65]: Expression yielded type double where int was expected.
[ExprTypesVisitor - 66]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 66]: Expression yielded type double where boolean was expected.
[ExprTypesVisitor - 67]: Expression yielded type int where double was expected.
[ExprTypesVisitor - 67]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 67]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 67]: Expression yielded type int where double was expected.
